react/ProductDetails.js

```diff
@@ -89,140 +89,178 @@ class ProductDetails extends Component {
       : (otherQuantity === 0 && -1) || price - otherPrice
   }
 
+  renderLoader() {
+    const { displayVertically } = this.props
+
+    const isVertical = displayVertically ? 'VERTICAL' : 'HORIZONTAL'
+
+    return (
+      <div className="vtex-product-details flex flex-wrap pa6">
+        <div className="vtex-product-details__images-container w-50-ns w-100-s pr5-ns">
+          <div className="fr-ns w-100 h-100">
+            <div className="flex justify-center">
+              <ProductImages.Loader isVertical={isVertical} />
+            </div>
+          </div>
+        </div>
+        <div className="vtex-product-details__details-container w-50-ns w-100-s pl5-ns">
+          <div className="fl-ns w-100">
+            <ProductDetails.Loader />
+          </div>
+        </div>
+      </div>
+    )
+  }
+
   render() {
     const { slug, displayVertically } = this.props
 
     return (
       <IntlInjector>
         {intl => (
-          <Query
-            query={productQuery}
-            variables={{ slug }}
-          >
+          <Query query={productQuery} variables={{ slug }}>
             {({ loading, data: { product } }) => {
-              if (loading || !product) {
-                return (
-                  <div className="pt6 tc">
-                    <Spinner />
-                  </div>
-                )
-              }
+              const shouldDisplayLoader = loading || !product
+              let selectedItem,
+                commertialOffer,
+                sellerId,
+                skuItems,
+                initialItemIndex
+              if (!shouldDisplayLoader) {
```

Comment:
wouldn't it be better if you just returned the loading state when `shouldDisplayLoader` is `true`? this way you'd avoid the need for the `let` in the variables declared above.

Reply:
This way i have less code and only one point of return